src/reference-manual/statements.qmd

```diff
@@ -365,42 +366,70 @@ To access accumulated log density up to the current execution point,
 
-The name "sampling statement" is meant to be suggestive, not
-interpreted literally.  Conceptually, the variable `y`, which may
-be an unknown parameter or known, modeled data, is being declared
-to have the distribution indicated by the right-hand side of the
-sampling statement. The above statement can read as "y is distributed
-as normal distribution with parameters mu and sigma.". The symbol `~`
-is called tilde.
+In general, we can read $\sim$ as "is distributed as," and overall this notation is used as a shorthand for defining distributions as
+$$
+\begin{aligned}
+   p(y| \mu, \sigma) & = \mathrm{normal}(y |  \mu, \sigma)\\
+   p(\mu) & = \mathrm{normal}(\mu |  0, 10)\\
+   p(\sigma) & = \mathrm{normal}^+(\sigma |  0, 1).
+\end{aligned}
+$$
+A collection of distribution statements define an unnormalized joint distribution as the product of component distributions
+$$
+p(y,\mu,\sigma) \propto p(y| \mu, \sigma )p(\mu) p(\sigma).
+$$
+In general, the product of arbitrary probability density functions is not a normalized probability density function---that is, it will be positive but will not in general integrate to 1---but the proportionality is sufficient for the Stan algorithms.
+
+Stan always constructs the target function---in Bayesian terms, the log posterior density function of the parameter vector---by adding terms in the model block.  Equivalently, each $\sim$ statement corresponds to a multiplicative factor in the unnormalized posterior density.
+
+This works even if the model is not constructed generatively.  For example, suppose you include the following code in a Stan model:
+```stan
+  a ~ normal(0, 1);
+  a ~ normal(0, 1);
+```
+This is translated to
+$$
+    p(a) = \mathrm{normal}(a | 0, 1)\mathrm{normal}(a |  0, 1),
+$$
+which in this case is $\mathrm{normal}(a|0,1/\sqrt{2})$.  One might expect that the above two lines of code would represent a redundant expression of a $\mathrm{normal}(a|0,1)$ prior, but, no, each line of code corresponds to an additional term in the target, or log posterior, density.  You can think of each line as representing an additional piece of information.
+
+Distribution statement `... ~ ...` accepts only distributions on the right side. These distributions can be built in or user defined distributions.  The left side of a distribution statement may be data, parameter, or a complex expression, but the evaluated type needs to match one of the allowed type of the right hand side distribution (see more below).
 
-Executing such a statement does not perform any sampling.  In Stan, a
-sampling statement is merely a notational convenience following the typical
+In Stan, a distribution statement is merely a notational convenience following the typical
 notation used to present models in the literature.  The above
-sampling statement could be expressed as a direct increment on the
+model defined with distribution statements could be expressed as a direct increment on the
```

Comment:
This produces a different log density function that includes normalizing terms.  I would suggest using `lupdf` so that they match.  Or at the very least, indicating that this is different and there's an lpdf/lupdf distinction in built-in (but not user-defined) distributions.